cli/src/commands/extensions/permissions.rs

```diff
@@ -0,0 +1,159 @@
+use std::path::{Component, PathBuf};
+
+use anyhow::{anyhow, Result};
+use deno_runtime::permissions::PermissionsOptions;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Default, Deserialize, Debug, Serialize)]
+pub struct Permissions {
+    read: Option<Vec<String>>,
+    write: Option<Vec<String>>,
+    run: Option<Vec<String>>,
+    net: Option<Vec<String>>,
+}
+
+// XXX In Deno, `Some(vec![])` actually means "allow all". We do not ever
+// want that, so we manually convert those instances into `None` in the
+// getter methods below. We need to make sure to always go through these when
+// constructing a `PermissionsOptions` object.
+impl Permissions {
+    pub fn read(&self) -> Option<&Vec<String>> {
+        self.read.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn write(&self) -> Option<&Vec<String>> {
+        self.write.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn run(&self) -> Option<&Vec<String>> {
+        self.run.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn net(&self) -> Option<&Vec<String>> {
+        self.net.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.read().is_none()
+            && self.write().is_none()
+            && self.run().is_none()
+            && self.net().is_none()
+    }
```

Comment:
The concept of "empty" for permissions seems strange to me. I'd probably go more the route of deno's `allow_all` method and call this `is_allow_none` or something like that. "Empty" could mean either everything allowed, or nothing allowed, it's not exactly clear.